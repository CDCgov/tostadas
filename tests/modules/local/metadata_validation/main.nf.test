nextflow_process {

    name "Test Process METADATA_VALIDATION"
    script "modules/local/metadata_validation/main.nf"
    process "METADATA_VALIDATION"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
                batch_size = 1
                project_dir = "${projectDir}" // (?: do I need this)
                output_dir = "${outputDir}"
                custom_fields_file = "${projectDir}/assets/example_custom_fields.json"
                validate_custom_fields = true
                date_format_flag = 's'
                remove_demographic_info = true
                validate_custom_fields = true
                fetch_reports_only = false
                resolved_output_dir = "${outputDir}" // (?: I'm not sure about this)
                resolved_submission_config = "${projectDir}/conf/submission_config.yaml"
                biosample_fields_key = "${projectDir}/assets/biosample_fields_key.yaml"
            }
            process {
                """
                // define inputs of the process here
                // input[0] = path("${projectDir}/assets/sample_metadata/mpxv_test_metadata.xlsx")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.tsv_files).match()
        }

    }

}
