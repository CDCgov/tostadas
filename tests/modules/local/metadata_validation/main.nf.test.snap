{
    "Should run without failures": {
        "content": [
            [
                [
                    "batch_1.tsv:md5,58f0cb49a968e687f1d6d6f19be76e11",
                    "batch_2.tsv:md5,c3379b86301ddc604364132292e1cf3a",
                    "batch_3.tsv:md5,8b02835d4e99c966ebf0d7ef3aaae429",
                    "batch_4.tsv:md5,e03be080faca211ad4da62b974c27626",
                    "batch_5.tsv:md5,06b2bc516d5fa8cfe20a992cec99dcac"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T16:22:04.094817068"
    }
}